src/delivery/controllers/roverControllers.js

```diff
@@ -0,0 +1,33 @@
+import axios from "axios";
+import { NASA_API_URL } from "../../config/index.js";
+import Exception from "../../utils/Exception.js";
+
+export const getRoverImage = async (req, res, next) => {
+  try {
+    const { userApiKey } = req.body;
+
+    const nasaApiUrl = `${NASA_API_URL}/mars-photos/api/v1/rovers/curiosity/latest_photos?api_key=${userApiKey}`;
+
+    const response = await axios.get(nasaApiUrl);
+
+    if (response.status !== 200 && !response?.data?.latest_photos?.length) {
+      return res.status(404).json({ message: "No photos found" });
```

Comment:
i that case you can catch error in controller catch and pass to the `next` middleware